presto-orc/src/main/java/com/facebook/presto/orc/reader/LongSelectiveStreamReader.java

```diff
@@ -73,12 +74,20 @@ public void startStripe(Stripe stripe)
                     directReader = new LongDirectSelectiveStreamReader(context);
                 }
                 currentReader = directReader;
+                if (dictionaryReader != null && context.isResetAllReaders()) {
```

Comment:
I think it would be better to close and nullify the other reader if it exists. Reader creation is not very expensive, and usually stripes stick with the same encoding.

Reply:
sure. will try that and run some test.